src/Symfony/Component/Security/Guard/Tests/Firewall/GuardAuthenticationListenerTest.php

```diff
@@ -184,6 +204,45 @@ public function testHandleCatchesAuthenticationException()
         $listener->handle($this->event);
     }
 
+    /**
+     * @legacy
+     */
+    public function testLegacyInterfaceAvoidSupportsCall()
+    {
+        $authenticatorA = $this->getMock('Symfony\Component\Security\Guard\GuardAuthenticatorInterface');
+        $providerKey = 'my_firewall3';
+
+        $authenticatorA
+            ->expects($this->never())
+            ->method('supports');
```

Comment:
see https://github.com/symfony/symfony/pull/18284/files#diff-d8949c4b7c75325b0bef282e1d2041b9R1235 for an example